src/main/java/com/limechain/runtime/hostapi/RuntimePointerSize.java

```diff
@@ -0,0 +1,18 @@
+package com.limechain.runtime.hostapi;
+
+public record RuntimePointerSize(long pointerSize) {
+    public RuntimePointerSize(int pointer, int size) {
+        this((long)size << 32 | (pointer & 0xffffffffL));
+    }
+
+    public RuntimePointerSize(Number pointerSize) {
+        this(pointerSize.longValue());
+    }
+    public int pointer() {
+        return (int) this.pointerSize;
+    }
+
+    public int size() {
+        return (int) (pointerSize >> 32);
+    }
```

Comment:
32 is a magic number. 
Always add more inline documentation when working with bitwise operators as the code is harder to read, `pointer & 0xffffffffL` should be explained.